File: src/NodeParser/StringLiteralNodeParser.ts
import ts from "typescript";
import type { Context } from "../NodeParser.js";
import type { SubNodeParser } from "../SubNodeParser.js";
import type { BaseType } from "../Type/BaseType.js";
import { LiteralType } from "../Type/LiteralType.js";

export class StringLiteralNodeParser implements SubNodeParser {
    public supportsNode(node: ts.StringLiteral): boolean {
        return node.kind === ts.SyntaxKind.StringLiteral;
    }
    public createType(node: ts.StringLiteral, context: Context): BaseType {
        return new LiteralType(node.text);
    }
}

File: internal/parser/string_literal_node_parser.go
package parser

import (
	"github.com/microsoft/typescript-go/shim/ast"

	"github.com/vega/ts-json-schema-generator/internal/types"
)

// StringLiteralNodeParser handles string literals
// (src/NodeParser/StringLiteralNodeParser.ts).
type StringLiteralNodeParser struct{}

func NewStringLiteralNodeParser() *StringLiteralNodeParser {
	return &StringLiteralNodeParser{}
}

func (p *StringLiteralNodeParser) SupportsNode(node *ast.Node) bool {
	return node.Kind == ast.KindStringLiteral
}

func (p *StringLiteralNodeParser) CreateType(node *ast.Node, _ *Context, _ *types.ReferenceType) types.Type {
	return &types.LiteralType{Value: node.Text()}
}
